docs/automations.md

```diff
@@ -0,0 +1,89 @@
+# Automations
+
+`foreman-js` uses github-actions to automate the test, build, and deployment of its packages.
+
+- [Testing](#testing)
+  * [Test `foreman-js`](#test--foreman-js-)
+  * [Test foreman and foreman plugins with `foreman-js` source code](#test-foreman-and-foreman-plugins-with--foreman-js--source-code)
+  * [Build tarballs](#build-tarballs)
+- [Deployment](#deployment)
+  * [Git and Github deployment](#git-and-github-deployment)
+  * [Deploy to npm](#deploy-to-npm)
+  * [Prereleases](#prereleases)
+
+
+# Testing
+
+The testing automation runs on every push and every pull request on Github.
+
+## Test `foreman-js`
+
+- Run code linting
+- Run commit linting
+- Run foreman-js testing
+- Run the build process to make sure there are no build errors
+
+## Test foreman and foreman plugins with `foreman-js` source code
+
+For `foreman` and the following plugins:
+- katello
+- foreman-tasks
+- foreman_rh_cloud
+
+Link to `foreman-js` source-code and run:
+- `npm run test`
+- `npm run lint`
+
+## Build tarballs
+
+Will build tarballs for each package.
+See: https://github.com/theforeman/foreman-js/blob/master/CONTRIBUTING.md#testing-pull-requests
+
+# Deployment
+
+The deployment process contain 2 steps:
+- [Deploy to github](#deploy-to-github)
+- [Deploy to npm](#deploy-to-npm)
+
+## Git and Github deployment
+
+Run on every push to the following branches:
+- `master`
+- `next**` (`next`, `next-react-v17`, `next-webpack-v4`)
```

Comment:
is it possible that we will have 2 identical npm libs names, e.g: `@theforeman/next-patternfly-upgrade`
how can we avoid that? I guess we won't be able to deploy to npm with an identical lib name, or will it be smart enough to bump its version?

Reply:
That should not happen because you will end up with different unique versions:
- 5.0.0-next.0
- 5.0.0-next.1
- 5.0.0-next.2
- 6.0.0-next.0
- 6.0.1-next.0